### PR TITLE
feat(action): always print x-deno-ray for debuggability

### PR DESCRIPTION
**Pull request by magurotuna**
This commit makes `x-deno-ray` values always printed in GitHub Actions as soon as we get a response from the backend API server.

The primary motivation behind it is that it's hard to debug issues with streaming APIs - in particular, `gitHubActionsDeploy`:

https://github.com/denoland/deployctl/blob/acb2ba33df6b146bc8705a1d1c16283d9fef9e78/src/utils/api.ts#L410-L424

This API returns 200 OK and then starts streaming the progress. But currently `x-deno-ray` is not printed when the status code is 200. Problems that happen during streaming, e.g. very slow file uploads, could be made easier to debug by making `x-deno-ray` available in this case too.

### CODE REVIEW

File: action/index.js [hunk 1]
```diff
@@ -86,7 +86,10 @@ async function main() {
   });
   core.debug(`Discovered ${assets.size} assets`);
 
-  const api = new API(`GitHubOIDC ${token}`, ORIGIN);
+  const api = new API(`GitHubOIDC ${token}`, ORIGIN, {
+    alwaysPrintDenoRay: true,
+    logger: core,
```
Comment (kt3k) [inline action/index.js:91]: I wonder if this need to be an option. How about just importing `"@actions/core"` in `src/utils/api.ts` and calling `core.notice()`?

File: action/index.js [hunk 2]
```diff
@@ -86,7 +86,10 @@ async function main() {
   });
   core.debug(`Discovered ${assets.size} assets`);
 
-  const api = new API(`GitHubOIDC ${token}`, ORIGIN);
+  const api = new API(`GitHubOIDC ${token}`, ORIGIN, {
+    alwaysPrintDenoRay: true,
```
Comment (kt3k) [inline action/index.js:90]: I also wonder this needs to be an option if it's always set to `true`

File: action/index.js (hunk 1 above)
Comment (magurotuna) [reply action/index.js:91]: `src/utils/api.ts` is called from not only GitHub Actions implementation but also deployctl (CLI) implementation. I wanted to keep `api.ts` agnostic about these differences.

File: action/index.js (hunk 2 above)
Comment (magurotuna) [reply action/index.js:90]: The intention here is that we don't turn on this option in deployctl CLI command.

File: action/index.js (hunk 1 above)
Comment (kt3k) [reply action/index.js:91]: Oh, makes sense

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**